pkg/trace/api/api_test.go

```diff
@@ -368,6 +368,31 @@ func TestReceiverDecodingError(t *testing.T) {
 	})
 }
 
+func TestReceiverUnexpectedEOF(t *testing.T) {
+	assert := assert.New(t)
+	conf := newTestReceiverConfig()
+	r := newTestReceiverFromConfig(conf)
+	server := httptest.NewServer(http.HandlerFunc(r.handleWithVersion(v05, r.handleTraces)))
+	// we get to read the header and the entire dictionary, but the request claims to be much larger
+	data := []byte{0x92, 0x91, 0xA5, 'a', 'b', 'c', 'd', 'e'}
+	client := &http.Client{}
+	traceCount := 2
+
+	t.Run("Content-Length larger than input", func(t *testing.T) {
```

Comment:
You don't need a sub-test here, because we only have 1 test.

Also, I want to point out a common misconception about Go sub-tests. In most languages the quoted string argument represents a description of what the unit test does. For Go, it's generally just one word and it merely serves as an identifier to be able to trace back the test when you see it failing in your output, and not as a description. If you want to dig deeper, feel free to grep the standard library for `t.Run` or for `*testing.T` to see what the common names and conventions are. One hint that something is wrong is running the tests with `go test -v` and observing the inserted underscores in place of spaces.

In this case I'd just recommend using `unexpectedEOF` or something, but as I've said above it's not needed here.